File: 2022/day01/day01-2.mjs
import * as fs from "node:fs";

function parseLineGroups() {
  const groups = fs.readFileSync("input.txt", "utf-8").split(/\n\s*\n/);
  return groups.map((input) => input.split(/\n/));
}
function sum(array) {
  return array.reduce((total, value) => total + value, 0);
}

const totals = parseLineGroups().map((group) => {
  const values = group.map((line) => parseInt(line, 10));
  return sum(values);
});

totals.sort((a, b) => b - a);

console.log("DEBUG totals:", totals);
console.log("RESULT:", sum(totals.slice(0, 3)));

File: 2022/day01/day01-2.py
with open('input.txt') as f:
    lines = f.read()

rawGroups = lines.split('\n\n')
groups = [[int(line) for line in line.split('\n')] for line in rawGroups]
groupSums = [sum(group) for group in groups]

groupSums.sort(reverse=True)
print("Sum Top 3: ", sum(groupSums[:3]))
